packages/thirdweb/src/extensions/erc721/drops/write/updateMetadata.ts

```diff
@@ -0,0 +1,123 @@
+import type { ThirdwebClient } from "../../../../client/client.js";
+import { prepareContractCall } from "../../../../transaction/prepare-contract-call.js";
+import type { BaseTransactionOptions } from "../../../../transaction/types.js";
+import type { NFT, NFTInput } from "../../../../utils/nft/parseNft.js";
+import { getBaseURICount } from "../../__generated__/IBatchMintMetadata/read/getBaseURICount.js";
+
+export type UpdateMetadataParams = {
+  targetTokenId: bigint;
+  newMetadata: NFTInput;
+  client: ThirdwebClient;
+};
+
+/**
+ * @internal
+ */
+export async function getUpdateMetadataParams(
+  options: BaseTransactionOptions<UpdateMetadataParams>,
+): Promise<[bigint, string]> {
+  const { contract, targetTokenId, newMetadata, client } = options;
+  const batchCount = await getBaseURICount(options);
+  if (batchCount === 0n) {
+    throw new Error(
+      "No base URI set. Please set a base URI before updating metadata",
+    );
+  }
+
+  // Look for the batchId & determine the start + end tokenId of the batch
+  const [{ getBatchIdAtIndex }, { getNFT }] = await Promise.all([
+    import("../../__generated__/IBatchMintMetadata/read/getBatchIdAtIndex.js"),
+    import("../../read/getNFT.js"),
+  ]);
+  let startTokenId = 0n;
+  let endTokenId = 0n;
+  let batchIndex = 0n;
+  for (let i = 0n; i < batchCount; i += 1n) {
+    batchIndex = i;
+    endTokenId = await getBatchIdAtIndex({ contract, index: batchIndex });
```

Comment:
this will block on every iteration, any way to do this in parallel?

Reply:
Yeah we can. I was thinking, what if there are 10,000 batches and the token ID we want to update is like, `0n`, then we are wasting resources.

But the chances of a drop collection having too many batches are low, so this "loop" way might provide not-too-bad performance, while saving on RPC requests.

wdyt @jnsdls

Reply:
So like, if I upload 10,000 NFTs at once, that's just 1 batch

Reply:
acutally i dont think i make sense anymore

Reply:
I think I follow, that somewhat makes sense this seems to just be pretty un-ergonomic on the contract side

Reply:
Ty. (yeah i have brought up this contract design a few times now)